benchmarks/profiler/utils/dynamo_deployment.py

```diff
@@ -77,6 +80,26 @@ def _init_kubernetes(self):
         self.custom_api = client.CustomObjectsApi(self.k8s_client)
         self.core_api = client.CoreV1Api(self.k8s_client)
 
+    def _is_running_in_kubernetes(self) -> bool:
+        """
+        Detect if we're running inside a Kubernetes cluster by checking for the service account token.
+        """
+        return os.path.exists("/var/run/secrets/kubernetes.io/serviceaccount/token")
+
+    def _get_service_url(self) -> str:
+        """
+        Get the service URL. Returns service DNS if running in Kubernetes, otherwise localhost with port forwarding.
+        """
+        if self._is_running_in_kubernetes():
+            # Use Kubernetes service DNS: service-name.namespace.svc.cluster.local:port
+            svc_name = f"{self.deployment_name}-frontend"
```

Comment:
This shouldn't block merge, but we should be cautious about hard-coding service names in this way. It's not necessarily a correct assumption.

An alternative would be to take this in via configuration, at least to override the default, and the operator can then change the value if it knows it deployed it with a different name.